content/post/la-jamstack-n-est-rapide-que-si-vous-la-rendez-rapide.md

```diff
@@ -0,0 +1,115 @@
+
+- La performance pour les utilisateurs est largement meilleure, avec du HTML déjà compilé sur le serveur et servi de manière statique.
+- Beaucoup moins d’appel d’API, requière beaucoup moins de temps de compilation et d’énergie.
+- Chacun devrait savoir que [Aaron Parecki](https://aaronparecki.com) propose l’impressionnant service [webmention.io](http://webmention.io) **gratuitement**, et la majorité des utilisateur de Webmention l’utiliser aujourd’hui, alors faites en sorte que son API se sente mieux.
+
+## Améliorer le côté client, s’il est indispensable
+
+Si vous savez que vous recevez beaucoup de webmentions très utiles que vous devez afficher à vos visiteurs, vous pouvez améliorer la liste générée côté serveur via le côté client.
+
+Mais rappelez-vous que chaque JavaScript ajouté à la page a un coût, donc les quelques webmentions supplémentaires doivent être vraiment utiles
+
+Alors, au lieu de faire ça sur cahque page vue, faites au moins :
+
+D’abord, essayez d’**attendre un peu après la génération du site** avant de faire les appels API côté client. Garder l’horodatage de génération du site côté client via JavaScript, et attendre une heure, une journée, en fonction de la fréquence des webmentions. Vous pouvez même utiliser l’« age » de la page pour moins requêter [webmention.io](http://webmention.io) pour le contenu plus ancien, qui reçoit probablement moins de webmentions, comme l’a fait [Aaron Gustafson pour les appels côté serveur dans son plugin Jekyll](https://aarongustafson.github.io/jekyll-webmention_io/performance-tuning).
+
+Ensuite, gardez les appels utilisateur à l’API, dans le *localStorage* ou l’*IndexDB*, afin que vous ne répétiez pas ces appels peu de temps après. Vous pouvez même utiliser un Service Worker pour mettre en cache les requêtes et leur horodatage.
+
+## Les appels à l’API uniquement côté client ont parfois plus de sens
+
+Je suis d’accord que les webmentions ne sont pas le cas d’usage le plus complexe pour expliquer que la plupart du temps vous devez appeler les API côté serveur au moment de la génération plutôt que côté client :
+
+- Les webmentions à afficher sont les mêmes pour tous les visiteurs.
+- Manquer quelques-unes des plus récentes n’est probablement pas un problème.
+
+Alors oui, de nombreux autres cas d’usage rendent les appels côté client nécessaires, ou meilleur que ceux côté serveur, je comprend ça.
+
+Ce que je dis c’est que **ça ne devrait pas être le cas par défaut**.
+
+## Promouvoir la ~~AJMstack~~ Mstack
+
+C’est aussi quelque chose que je n’aime pas vraiment dans la tendance actuelle de la JAMstack, promouvoir **J**avaScript et les **A**PI bien plus que le balisage (NDT : « balisage » peut être traduit « **M**arkup » en anglais).
+
+Voici pour l’exemple ce que vous pouvez voir sur [jamstack.wtf](https://jamstack.wtf/) (simplifié) :
+
+![JAMstack-schema-1](/assets/images/jamstack/JAMstack-schema-1.png)
```

Comment:
@Narno @DirtyF vous préférez en images plutôt qu'avec HTML/CSS ?

Reply:
@nhoizey je pense que ce serait plus cohérent avec l'article original de rendre les schéma en HTMl/CSS, mais j'ai pas pris le temps de me pencher dessus ce soir : les images sont des "placeholders".

Reply:
Ok !

Voilà les styles dédiés à ces schémas :
https://nicolas-hoizey.com/articles/2020/05/05/jamstack-is-fast-only-if-you-make-it-so/styles.css

Reply:
@nhoizey done.

Reply:
On a finalement opté pour du SVG pour se simplifier l'intégration.

Reply:
SVG plus simple que HTML/CSS ? 😅

Reply:
Ha ha, oui : j'avais effectué une 1ère intégration avec ton CSS, mais il y avait un léger décalage (collusion avec d'autres styles ?) et Frank m'a proposé d'en faire des SVG ;-)

Reply:
ok 😅